### ISSUE TITLE
VM returns different result than Prometheus

### ISSUE THREAD

**Issue by phantasia15** [VictoriaMetrics/VictoriaMetrics#577]
**Describe the bug**
group_left(labels) seems to take labels from the "many" side while it should take from the "one" side

**To Reproduce**
```
label_replace(
                kube_pod_owner{job="kube-state-metrics", owner_kind="ReplicaSet"},
                "replicaset", "$1", "owner_name", "(.*)"
 ) * on(replicaset, namespace) group_left(owner_name) kube_replicaset_owner{job="kube-state-metrics"}
```
VM currently returns wrong `owner_name` value:
```
{namespace=dev, pod=flight-9df5c6f9c-7dgkf, owner_name=flight-9df5c6f9c}
```
**Expected behavior**
Basically, both the left & right side of the binary operator contain "owner_name" label but with different values. Since I specify `group_left(owner_name)`, the `owner_name` of the right side should take priority in the output.

Here is the result of the left side:
```
{namespace=dev, pod=flight-9df5c6f9c-7dgkf, owner_name=flight-9df5c6f9c, owner_kind=Replicaset, replicaset=flight-9df5c6f9c}
```
The result of the right side:
```
{namespace=dev, replicaset=flight-9df5c6f9c, owner_name=flight, owner_kind=Deployment}
```

The Prometheus docs [here](https://prometheus.io/docs/prometheus/latest/querying/operators/#many-to-one-and-one-to-many-vector-matches) also states that the labels in group_left(labels) should be taken from "one" side.  So VM should return this instead (it is also what Prometheus returns):
```
{namespace=dev, pod=flight-9df5c6f9c-7dgkf, owner_name=flight}
```

**Version**
The line returned when passing `--version` command line flag to binary. 
```
victoria-metrics-20200620-223134-tags-v1.37.2-0-ge149019c0
```

**Comment by valyala** [comment]
Confirmed that VictoriaMetrics doesn't override `oner_name` label value from left side with the value from right. This differs from Prometheus behavior, which can be inspected by the following links:

* [empty group_left list](http://demo.robustperception.io:9090/graph?g0.range_input=1h&g0.expr=label_replace(%0A%20%20label_replace(%0A%20%20%20%20label_replace(%0A%20%20%20%20%20%20label_replace(vector(1)%2C%20%22owner_name%22%2C%20%22flight-123%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%20%20%20%20%22pod%22%2C%20%22flight-aaa%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%20%20%22replicaset%22%2C%20%22flight-123%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%22namespace%22%2C%20%22dev%22%2C%20%22%22%2C%20%22.*%22)%20*%20on(replicaset%2C%20namespace)%20group_left%0Alabel_replace(%0A%20%20label_replace(%0A%20%20%20%20label_replace(vector(1)%2C%20%22owner_name%22%2C%20%22flight%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%20%20%22replicaset%22%2C%20%22flight-123%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%22namespace%22%2C%20%22dev%22%2C%20%22%22%2C%20%22.*%22)&g0.tab=1) - Prometheus uses `owner_name` from the left side
* [group_left(owner_name)](http://demo.robustperception.io:9090/graph?g0.range_input=1h&g0.expr=label_replace(%0A%20%20label_replace(%0A%20%20%20%20label_replace(%0A%20%20%20%20%20%20label_replace(vector(1)%2C%20%22owner_name%22%2C%20%22flight-123%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%20%20%20%20%22pod%22%2C%20%22flight-aaa%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%20%20%22replicaset%22%2C%20%22flight-123%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%22namespace%22%2C%20%22dev%22%2C%20%22%22%2C%20%22.*%22)%20*%20on(replicaset%2C%20namespace)%20group_left(owner_name)%0Alabel_replace(%0A%20%20label_replace(%0A%20%20%20%20label_replace(vector(1)%2C%20%22owner_name%22%2C%20%22flight%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%20%20%22replicaset%22%2C%20%22flight-123%22%2C%20%22%22%2C%20%22.*%22)%2C%0A%20%20%22namespace%22%2C%20%22dev%22%2C%20%22%22%2C%20%22.*%22)&g0.tab=1) - Prometheus uses `owner_name` from the right side.

This bug will be fixed soon.

**Comment by valyala** [comment]
The bug should be fixed in the commit 7532dbcdf54439fb33a66d9e9957f93072abf745 . @phantasia15 , could you buld VictoriaMetrics from this commit according to [the instructions](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#how-to-build-from-sources) and verify whether the bug is fixed there?

**Comment by phantasia15** [comment]
@valyala Confirmed that the bug is fixed with the above commit.
Thank you very much!

**Comment by valyala** [comment]
@phantasia15 , thanks for the confirmation.

Re-opening the issue until the bugfix is included in the next release.

**Comment by valyala** [comment]
The bugfix has been included in [v1.37.3](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
